integration-tests/sanity-functional-protractor/specs/api-spec.js

```diff
@@ -0,0 +1,121 @@
+            const parametersCorrelationId = testDataJson.get_metadata_by_correlationId.parameters;
+            const jsonSchemaPath = require(testDataJson.get_metadata_by_correlationId.json_schema_path);
+            return AetsApi.getMetadata(parametersSuiteName).then((response) => {
+                parametersCorrelationId.correlationId = response.body.correlationId;
+            }).then(() => {
+                return AetsApi.getMetadata(parametersCorrelationId).then((response) => {
+                    chakramExpect(response).to.have.status(HTTP_200_OK);
+                    chakramExpect(response).to.have.schema(jsonSchemaPath);
+                    jasmineExpect(response.body.correlationId).toEqual(parametersCorrelationId.correlationId);
+                });
+            });
+        });
+
+        it('Error message validation', () => {
+            const parametersCorrelationId = clone(testDataJson.get_metadata_by_correlationId.parameters);
+            parametersCorrelationId.correlationId = "no_existing_correlationId"
+            const errorMessageJson = testDataJson.get_metadata_by_correlationId.error_message;
+            return AetsApi.getMetadata(parametersCorrelationId).then((response) => {
+                chakramExpect(response).to.have.status(HTTP_400_BAD_REQUEST);
+                jasmineExpect(response.body).toEqual(errorMessageJson);
+            });
+        });
+    });
+
+    describe('Get artifact by artifact Id - validation', () => {
+
+        it('Status validation', () => {
+            const parametersSuiteName = testDataJson.get_metadata_by_suite_name.parameters;
+            const parametersArtifactId = testDataJson.get_artifact_by_artifact_Id.parameters;
+            return AetsApi.getMetadata(parametersSuiteName).then((response) => {
+                parametersArtifactId.id = recursiveKeySearch("artifactId", response.body)[1];
+            }).then(() => {
+                return AetsApi.getArtifact(parametersArtifactId).then((response) => {
+                    chakramExpect(response).to.have.status(HTTP_200_OK);
+                });
+            });
+        });
+
+        it('Error message validation', () => {
+            const errorMessageJson = testDataJson.get_metadata_by_correlationId.error_message;
+            const parametersCorrelationId = clone(testDataJson.get_metadata_by_correlationId.parameters);
+            parametersCorrelationId.correlationId = "no_existing_correlationId"
+            return AetsApi.getMetadata(parametersCorrelationId).then((response) => {
+                chakramExpect(response).to.have.status(HTTP_400_BAD_REQUEST);
+                jasmineExpect(response.body).toEqual(errorMessageJson);
+            });
+        });
+    });
+});
+
+this.jsonSearchForKey = (jsonObj, keyName) => {
+    for (let key in jsonObj) {
+        if (key == keyName) return key;
+        else this.jsonSearchForKey(key, keyName);
+    }
+};
+
+const recursiveKeySearch = (key, data) => {
```

Comment:
All of the code below looks like helpers, thus it would be better to move this to another file, and if not at least it is better to put them at the top of the page. Actually ES6 won't allow you to use const in the script before they are declared.

Reply:
done